app/models/grant.rb

```diff
@@ -1,12 +1,19 @@
-  # frozen_string_literal: true
+# frozen_string_literal: true
 
 class Grant < ApplicationRecord
   include ActiveModel::Validations::Callbacks
-  include SoftDeletable
+
   extend FriendlyId
   has_paper_trail versions: { class_name: 'PaperTrail::GrantVersion' }
   friendly_id :slug
 
+  include Discard::Model
+
+  after_discard do
```

Comment:
what about after_undiscard? shoudln't it restore the associated records too?

Reply:
yes. added in latest commit.